net/netifyd/files/netifyd.init

```diff
@@ -11,39 +11,45 @@ USE_PROCD=1
 PROG=/usr/sbin/netifyd
 
 start_netifyd() {
-	local instance enabled autoconfig internal_if external_if
-
-	instance="$1"
-
-	config_get_bool enabled "$instance" enabled 0
-	[ "$enabled" -eq 0 ] && return 0
-
-	[ ! -d /var/run/netifyd ] && mkdir -p /var/run/netifyd
-
-	config_get_bool autoconfig "$instance" autoconfig 1
-	config_get filter "$instance" filter
-
-	if [ "$autoconfig" -gt 0 ] ; then
-		source /usr/share/netifyd/functions.sh
-		load_modules
-		NETIFYD_AUTODETECT=yes
-		NETIFYD_OPTS=$(auto_detect_options)
-	else
-		config_get internal_if "$instance" internal_if "eth0"
-		config_get external_if "$instance" external_if "eth1"
-		NETIFYD_OPTS="-E $external_if -I $internal_if"
-	fi
-
-
-	procd_open_instance
-	procd_set_param command $PROG -R $NETIFYD_OPTS
-	[ -n "$filter" ] && procd_append_param command -F "$filter"
-	procd_set_param file /etc/netifyd.conf
-	procd_set_param respawn
-	procd_close_instance
+    local autoconfig enabled external_if instance interfaces internal_if options
+
+    instance="$1"
+
+    source /usr/share/netifyd/functions.sh
+    load_modules
+
+    config_get_bool enabled "$instance" enabled 0
+    [ "$enabled" -eq 0 ] && return 0
+
+    [ ! -d /var/run/netifyd ] && mkdir -p /var/run/netifyd
+
+    config_get_bool autoconfig "$instance" autoconfig 1
+
+    if [ "$autoconfig" -gt 0 ]; then
+        NETIFYD_AUTODETECT=yes
+        interfaces="$(auto_detect_options)"
+    else
+        config_get internal_if "$instance" internal_if ""
+        for ifn in $internal_if; do
+            interfaces="${interfaces} -I ${ifn}"
+        done
+        config_get external_if "$instance" external_if ""
+        for ifn in $external_if; do
+            interfaces="${interfaces} -E ${ifn}"
+        done
+    fi
+
+    config_get_bool options "$instance" options ""
```

Comment:
I suspect you mean config_get since options is a string

Reply:
Good catch!

Fixed.

Also, updated to use lists for internal/external_if entries, and options.

Anything obviously wrong with these changes?

Reply:
Nothing screams, in fact the updated options have explained the '-F' syntax to me and why I was getting such confusing results.  I'll update my test build and let you know if anything leaps out at me.

At the risk of stylistic war, tabs preferred to spaces and a tab=8.  Tab = 1 byte vs 4/8 spaces = 4/8 bytes.  The config & init script could do with that change basically to match everything else.

Reply:
Also L77 of Makefile - I don't think you need the CONFIGURE_ARGS parameter since it is included by default https://git.openwrt.org/?p=openwrt/openwrt.git;a=blob;f=include/package-defaults.mk#l120

$(call Build/Configure/Default) should be sufficient, or it seems to work for me

Reply:
Thanks for the feedback, changes committed.